galaxy_ng/app/settings.py

```diff
@@ -190,3 +191,96 @@
+    KEYCLOAK_HOST_LOOPBACK = os.getenv('KEYCLOAK_HOST_LOOPBACK', default=None)
+    KEYCLOAK_URL = "{protocol}://{host}:{port}".format(protocol=KEYCLOAK_PROTOCOL,
+                                                       host=KEYCLOAK_HOST, port=KEYCLOAK_PORT)
+    auth_url_str = "{keycloak}/auth/realms/{realm}/protocol/openid-connect/auth/"
+    SOCIAL_AUTH_KEYCLOAK_AUTHORIZATION_URL = auth_url_str.format(keycloak=KEYCLOAK_URL,
+                                                                 realm=KEYCLOAK_REALM)
+    if KEYCLOAK_HOST_LOOPBACK:
+        loopback_url = "{protocol}://{host}:{port}".format(protocol=KEYCLOAK_PROTOCOL,
+                                                           host=KEYCLOAK_HOST_LOOPBACK,
+                                                           port=KEYCLOAK_PORT)
+        SOCIAL_AUTH_KEYCLOAK_AUTHORIZATION_URL = auth_url_str.format(keycloak=loopback_url,
+                                                                     realm=KEYCLOAK_REALM)
+    token_url_str = "{keycloak}/auth/realms/{realm}/protocol/openid-connect/token/"
+    SOCIAL_AUTH_KEYCLOAK_ACCESS_TOKEN_URL = token_url_str.format(keycloak=KEYCLOAK_URL,
+                                                                 realm=KEYCLOAK_REALM)
+
+    SOCIAL_AUTH_LOGIN_REDIRECT_URL = os.getenv('SOCIAL_AUTH_LOGIN_REDIRECT_URL', default='/ui/')
+    SOCIAL_AUTH_POSTGRES_JSONFIELD = True
+    SOCIAL_AUTH_JSONFIELD_CUSTOM = 'django.contrib.postgres.fields.JSONField'
+    SOCIAL_AUTH_URL_NAMESPACE = 'social'
+    SOCIAL_AUTH_KEYCLOAK_EXTRA_DATA = [('refresh_token', 'refresh_token'),
+                                       (KEYCLOAK_ROLE_TOKEN_CLAIM, KEYCLOAK_ROLE_TOKEN_CLAIM), ]
+
+    SOCIAL_AUTH_PIPELINE = (
+        'social_core.pipeline.social_auth.social_details',
+        'social_core.pipeline.social_auth.social_uid',
+        'social_core.pipeline.social_auth.social_user',
+        'social_core.pipeline.user.get_username',
+        'social_core.pipeline.social_auth.associate_by_email',
+        'social_core.pipeline.user.create_user',
+        'social_core.pipeline.social_auth.associate_user',
+        'social_core.pipeline.social_auth.load_extra_data',
+        'social_core.pipeline.user.user_details',
+        'social_core.pipeline.debug.debug',
+        'galaxy_ng.app.pipelines.user_role',
+        'galaxy_ng.app.pipelines.user_group',
+    )
+
+    # Set external authentication feature flag
+    GALAXY_FEATURE_FLAGS["external_authentication"] = True
+
+    # Add to installed apps
+    INSTALLED_APPS.append('social_django')
+
+    # Add to authentication backends
+    AUTHENTICATION_BACKENDS = [
+        # BEGIN: Pulp standard authentication backends
+        "django.contrib.auth.backends.ModelBackend",
+        "guardian.backends.ObjectPermissionBackend",
+        # END: Pulp standard authentication backends
+        'social_core.backends.keycloak.KeycloakOAuth2',
+    ]
```

Comment:
```suggestion
    AUTHENTICATION_BACKENDS = [
        'social_core.backends.keycloak.KeycloakOAuth2',
        'dynaconf_merge'
    ]
```
If a list already exists in the main settings (which does [here](https://github.com/pulp/pulpcore/blob/master/pulpcore/app/settings.py#L118)) then you can contribute to the existing list by adding a `'dynaconf_merge'` to the list, this way dynaconf will append your items keeping the existing and if pulp changes this list in the future their changes will reflect.

There is actually another simpler syntax to achieve the same but the previous style is adopted here

```py
AUTHENTICATION_BACKENDS="@merge 'social_core.backends.keycloak.KeycloakOAuth2'"
```

Reply:
updated.